slurm-mlr/R/my-functions.R

```diff
@@ -0,0 +1,30 @@
+#' @title mlr::benchmark() wrapper
+#'
+#' @param learners
+#' @param tasks
+#' @param resampling
+#' @param measures
+#' @param cpus
+
+benchmark_custom <- function(learners, tasks, resampling, measures, cpus) {
+
+  # `mode = "multicore"` only works on Unix
+  parallelStart(
+    mode = "multicore", cpus = cpus, level = "mlr.resample",
+    mc.set.seed = FALSE
+  )
+  set.seed(12345)
```

Comment:
Is this part of the workaround for `make(lock_envir = TRUE)` + `parallel`? If so, should we really set the same seed for different targets that call `benchmark_custom()`?

Reply:
> Is this part of the workaround for make(lock_envir = TRUE) + parallel?

Not a workaround, its just the only way in mlr to use parallelization. It uses the `parallelMap` pkg in the background. `mode = "multicore` in fact uses `mclapply` in the background but I haven't faced any problems yet.

> should we really set the same seed for different targets that call benchmark_custom()?

Yes this is very important as the seeds affects the resampling split and these should be the same across different benchmark settings to make them comparable.

Reply:
I see, that makes sense now. We want to compare those different methods on the same data. Would you add a comment next to `set.seed(12345)` to clarify?